src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1119,7 +1120,7 @@ static ErrorCode getRefMismatchErrorCode(TypeKind type)
             }
 
             ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, selectedMethod, location, isDelegateConversion: true);
-
+            ReportDiagnosticsIfObsolete(diagnostics, selectedMethod, syntax, hasBaseReceiver: false);
```

Comment:
>ReportDiagnosticsIfObsolete [](start = 12, length = 27)

Should this check occur before the unsafe checks above? For instance, are we reporting an obsolete error for `M1()` below?
```C#
using System;
unsafe delegate byte* D();
class C
{
    [Obsolete(null, true)] unsafe static byte* F() => default;
    unsafe static D M1() => new D(F);
    static D M2() => new D(F);
}
```

Reply:
I don't think the warning is currently reported if unsafe not allowed error is reported. This should apply to UnmanagedCallersOnly warnings too.

I'll move both of them.